nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -69,6 +98,41 @@ void LatticeMotionTable::initMotionModel(
 
   // TODO(Matt) populate num_angle_quantization, size_x, min_turning_radius, trig_values,
   // all of the member variables of LatticeMotionTable
+
+  num_angle_quantization = latticeMetadata.number_of_headings;
+  num_angle_quantization_float = static_cast<float>(num_angle_quantization);
+  min_turning_radius = latticeMetadata.min_turning_radius;
+
+  //Find the largest turning radius that is not equal to zero
+  float largestTurningRadius = std::numeric_limits<float>::max();
+  for(auto const& motionPrimitivesAtAngle : motionPrimitives)
+  {
+    for(auto const& motionPrimitive : motionPrimitivesAtAngle)
+    {
+      if( motionPrimitive.radius < largestTurningRadius && motionPrimitive.radius != 0 )
+      {
+        largestTurningRadius = motionPrimitive.radius;
+      }
+    }
+  }
+
+  //NOTE: Note sure what type of state space this should be 
+  state_space = std::make_unique<ompl::base::DubinsStateSpace>( largestTurningRadius );
+
+  for(auto const& motionPrimitivesAtAngle : motionPrimitives)
+  {
+    for(auto const& motionPrimitive : motionPrimitivesAtAngle)
+    {
+      //TODO: angles in json file need to be converted to radians
+      primitive_start_angles.emplace_back( motionPrimitive.start_angle );
+    }
+  }
+
+  //TODO: angles in json file need to be converted to radians 
+  for(auto const &start_angle : primitive_start_angles)
+  {
+    trig_values.emplace_back( cos(start_angle), sin(start_angle) );
+  }
 }
 
 MotionPoses LatticeMotionTable::getProjections(const NodeLattice * node)
```

Comment:
So we have this function getProjections which is what we use to return the set of primitives that can be expanded from the given `node`. In the Hybrid-A* node, we use the `projections` variable but it looks like you instead populate a `motionPrimitives` object that meets basically the same thing.

So my recommendation would be to remove the `projections` variable since you're not using it and you can rename that function `getMotionPrimitives` to use your `motionPrimitives` object instead. You can now populate this function `getProjections` since you have all the information required. The starting node (so you know your starting heading) and the vector of motion primitives at each starting heading to return a `MotionPoses` of primitives. 

If you find that its hard to convert from motionPrimitives to the current `MotionPoses` object, you are welcome to replace `MotionPoses` with a new struct.

Reply:
You can also populate the `getLatticeMetadata` function as well (and even use that function in your `init` functions too!)

Reply:
Done, I left a few questions in the code